File: wenjun_liu/week_01/day_05/subway/line.js
class Line {
  constructor(name, stations){
    this.name = name;
    this.stations = stations;
  };

  checkLine(line){
    return this.name === line;
  };

  findStationIndex(stationName){
    for (let i=0;i<this.stations.length;i++){
      if(this.stations[i].name === stationName){
        return i;
      }
    }
  };

  makeTrip(stationFrom, stationTo){
    let fromIndex = this.stations.findIndex(station => station.name === stationFrom.name);
    let toIndex = this.stations.findIndex(station => station.name === stationTo.name);
    const reverse = fromIndex > toIndex;
    if(reverse){
      [fromIndex, toIndex] = [toIndex, fromIndex];
    }
    const trip = this.stations.slice(fromIndex+1, toIndex+1);
    return reverse ? trip.reverse() : trip;
  }
};

File: wenjun_liu/week_04/day_02/line.rb
class Line
  def initialize(name, stations)
    @name = name
    @stations = stations
  end

  def check_line(line)
    @name == line
  end

  def stations
    @stations
  end

  def find_station_index(station_name)
    @stations.each do |s|
      if s.name == station_name
          return @stations.index(s)
      end
    end
  end

  def make_trip(station_from, station_to)
    from_index = @stations.index{|s|s.name == station_from.name}
    to_index = @stations.index{|s|s.name == station_to.name}
    reverse = from_index > to_index
    if reverse
      from_index, to_index = to_index, from_index
    end
    trip = @stations[from_index..to_index]
    reverse ? trip.reverse : trip;
  end
end
